lib/linalg/BasisGenerator.cpp

```diff
@@ -129,35 +127,31 @@ BasisGenerator::isNextSample(
 bool
 BasisGenerator::takeSample(
     double* u_in,
-    double time,
-    double dt,
     bool add_without_increase)
 {
     CAROM_VERIFY(u_in != 0);
-    CAROM_VERIFY(time >= 0);
+    CAROM_VERIFY(d_svd->getNumSamples() < d_svd->getMaxNumSamples());
 
     // Check that u_in is not non-zero.
     Vector u_vec(u_in, getDim(), true);
     if (u_vec.norm() == 0.0) {
-        printf("WARNING: BasisGenerator::takeSample skipped trivial sample at time %.4E\n",
-               time);
+        printf("WARNING: BasisGenerator::takeSample skipped trivial sample.\n");
         return false;
     }
 
-    if (getNumBasisTimeIntervals() > 0 &&
-            d_svd->isNewTimeInterval()) {
-        resetDt(dt);
-        if (d_basis_writer) {
-            if (d_write_snapshots) {
-                writeSnapshot();
-            }
-            else {
-                d_basis_writer->writeBasis("basis");
-            }
-        }
-    }
-
-    return d_svd->takeSample(u_in, time, add_without_increase);
+    /*
+        Note for previous implementation:
```

Comment:
I think it is better not to document previous implementations. It is simpler just to remove `resetDt`.